src/app/content/utils/seoUtils.ts

```diff
@@ -0,0 +1,20 @@
+import { assertDefined } from '../../utils';
+import { Book, Page } from '../types';
+import { archiveTreeSectionIsChapter, findArchiveTreeNode, getArchiveTreeSectionNumber } from './archiveTreeUtils';
+
+export const createTitle = (page: Page, book: Book): string => {
+  let node = assertDefined(findArchiveTreeNode(book.tree, page.id), `couldn't find node for a page id: ${page.id}`);
+  const domNode = new DOMParser().parseFromString(node.title, 'text/html');
+  const osText = domNode.querySelector('.os-text');
+  const nodeTitle = osText ? osText.textContent : page.title;
+  let title = `${nodeTitle} - ${book.title} | OpenStax`;
+  while (node && node.parent) {
+    node = node.parent;
```

Comment:
I'm not exactly sure of this but don't we want to start searching for the number from the node itself not its parent?

Reply:
No, we want `Ch. [#][Page Title] - [Book] | OpenStax` if `page` is inside a `chapter` and `[Page Title] - [Book] | OpenStax` in other cases